tests/test_configuration.py

```diff
@@ -0,0 +1,45 @@
+from unittest import TestCase
+
+from nose.tools import eq_, ok_, assert_raises
+
+from ddtrace import config as global_config
+from ddtrace.configuration import Config, ConfigException
+
+
+class ConfigTestCase(TestCase):
+    """Test the `Configuration` class that stores integration settings"""
+    def setUp(self):
+        self.config = Config()
+
+    def test_registration(self):
+        # ensure an integration can register a new list of settings
+        settings = {
+            'distributed_tracing': True,
+        }
+        self.config._add('requests', settings)
+        ok_(self.config.requests['distributed_tracing'] is True)
+
+    def test_settings_copy(self):
+        # ensure that once an integration is registered, a copy
+        # of the settings is stored to avoid side-effects
+        settings = {
+            'distributed_tracing': True,
+        }
```

Comment:
if this is a real use case, I'd test the copy on 
```
settings = {
  'distributed_tracing': True,
  'foo': ['bar', 'baz'],
}

Reply:
Sure will update that.